### ISSUE TITLE
gmt meca -Sm throws error if Mrr=0

### ISSUE THREAD

**Issue by tomeracles** [GenericMappingTools/gmt#4225]
**Description of the problem**

When Mrr=0 (as is the case in some solutions in the Harvard CMT catalogue), gmt meca -Sm breaks and doesn't plot the focal mechanism. I get an error that "Assertion failed: (n < 2)" in ../src/seis/utilmeca.c, line 714. This is a problem in GMT >= 6.1.0, but not a problem in GMT 6.0.0. If I change Mrr to 0.01 then the mechanism plots fine. I don't think this is a problem for any other of the components of the moment tensor.

**Full script that generated the error**

```
gmt meca -Sm0.5c -JM5c -R20/30/35/45  -Gred -W1 -T0 -pdf test1 << EOF
25.06 38.38 15 -0.00 1.08 -1.08 -0.75 -0.07 -0.46 24 X Y
EOF
```

**Full error message**

```
Assertion failed: (n < 2), function meca_ps_tensor, file ../src/seis/utilmeca.c, line 714.
Abort trap: 6
```

**Actual outcome**

nothing

**Expected outcome**

![image](https://user-images.githubusercontent.com/49723355/93485780-5a149700-f8fb-11ea-980f-9e7a5e396024.png)

Cheers
Tom

**System information**

* Operating system: macOS Mojave 10.14.1
* GMT version (`gmt --version`): 6.1.1

**Comment by seisman** [comment]
I can confirm that there must be a bug.

Running the example above, I don't see the same error. Instead, I have a beachball like this, which still doesn't make sense. I will do some debugging later.
![image](https://user-images.githubusercontent.com/3974108/93520784-07d47580-f8fd-11ea-9a2d-8b7858aedb8f.png)

BTW, how do you install GMT 6.1.1 on macOS? Are you using the macOS bundle, homebrew or macports?

**Comment by tomeracles** [comment]
I installed the macOS bundle (dmg file). Same with GMT 6.1.0 - same issue. However, we also found this issue on a Linux machine we use for remote work for students, where the GMT version is  6.1.0_bb5dd8f_2019.10.10. I don't know how the IT wizards installed this version.

Thanks!

**Comment by seisman** [comment]
Thanks for your feedback. I'll see if I can fix it.

**Comment by tomeracles** [comment]
I have an update in case it helps - this error does not occur using GMT 6.0.0 on Mac, but it _does_ occur using GMT 6.0.0 when we installed it on Linux.

**Comment by tomeracles** [comment]
BTW we've encountered the same error message for some CMT solutions using gmt coupe, which must come from the same bug in psmeca (not to do with Mrr this time but I don't know what coupe does to the components before calling psmeca).

```
$ gmt coupe -JX10i/-4i -Bxf50a100 -Byf10a50 -BWeSn -R-117/-60/-48/10 -Aa-73.08/-24/-61.5/-23.53/90/100/0/1000k+f -Sm0.4c/-1 -Gred -W1 -T0 -Q -pdf test2 << EOF
> -69.02 -23.98 107 -2.08 -0.01 2.09 1.24 -1.27 1.59 23 X Y 201402232340A
> EOF
Assertion failed: (n < 2), function meca_ps_tensor, file ../src/seis/utilmeca.c, line 714.
Abort trap: 6
```

Again, on macOS there is no error if using 6.0.0 but there is using 6.1.x.

Cheers

**Comment by seisman** [comment]
@tomeracles Thanks for reporting the issue. It should be fixed by PR #4300.

**Comment by seisman** [comment]
@tomeracles PR #4300 has been merged into master branch. It should work well for you if you build GMT from master branch source codes.

Please let me know if there are any other failing beachballs.
